### PR TITLE
[FLINK-17810][doc] Add document for K8s application mode

### PR DESCRIPTION
**Pull request by wangyang0918**
Add document for K8s application mode.
* How to start a Flink application
* How to stop a Flink application

### CODE REVIEW

File: docs/ops/deployment/native_kubernetes.md [hunk 1]
```diff
@@ -170,6 +170,34 @@ appender.console.layout.pattern = %d{yyyy-MM-dd HH:mm:ss,SSS} %-5p %-60c %x - %m
...
+  -Dkubernetes.cluster-id=<ClusterId> \
+  -Dtaskmanager.memory.process.size=4096m \
+  -Dkubernetes.taskmanager.cpu=2 \
+  -Dtaskmanager.numberOfTaskSlots=4 \
+  -Dkubernetes.container.image=<custom_image_name>
+  local:///opt/flink/examples/streaming/WindowJoin.jar
+{% endhighlight %}
+
+Note: When deploying a Flink application, the user main code will be executed on Flink Master in the cluster. So the user jars and dependencies should already exist in the image. Follow the [how to customize the Flink Docker image](docker.html#customize-flink-image) to build a customized image baked in user jars on top of Flink official image.
+
```
Comment (sjwiesman) [inline docs/ops/deployment/native_kubernetes.md:189]: This needs to go on line 176 above the bash command. I'd also like to show a minimal dockerfile example. 

```suggestion
Application mode allows users to create a single image containing their Job and the Flink runtime, which will automatically create and destroy cluster components as needed. The Flink community provides base docker images [customized](docker.html#customize-flink-image) for any use case.

{% highlight dockerfile %}
FROM flink

COPY /target/my-flink-job-*.jar /opt/flink/usrlib/my-flink-job.jar
{% endhighlight %}

```

File: docs/ops/deployment/native_kubernetes.md [hunk 2]
```diff
@@ -170,6 +170,34 @@ appender.console.layout.pattern = %d{yyyy-MM-dd HH:mm:ss,SSS} %-5p %-60c %x - %m
...
+$ ./bin/flink run-application -p 8 -t kubernetes-application \
+  -Dkubernetes.cluster-id=<ClusterId> \
+  -Dtaskmanager.memory.process.size=4096m \
+  -Dkubernetes.taskmanager.cpu=2 \
+  -Dtaskmanager.numberOfTaskSlots=4 \
+  -Dkubernetes.container.image=<custom_image_name> \
+  local:///opt/flink/examples/streaming/WindowJoin.jar
+{% endhighlight %}
+
+Note: When deploying a Flink application, the user main code will be executed on Flink Master in the cluster. So the user jars and dependencies should already exist in the image. Follow the [how to customize the Flink Docker image](docker.html#customize-flink-image) to build a customized image baked in user jars on top of Flink official image.
```
Comment (sjwiesman) [inline docs/ops/deployment/native_kubernetes.md:188]: This needs to go on line 176 above the bash command. I'd also like to show a minimal dockerfile example. 

```suggestion
Application mode allows users to create a single image containing their Job and the Flink runtime, which will automatically create and destroy cluster components as needed. The Flink community provides base docker images [customized](docker.html#customize-flink-image) for any use case.

{% highlight dockerfile %}
FROM flink
COPY /target/my-flink-job-*.jar $FLINK_HOME/usrlib/my-flink-job.jar
{% endhighlight %}
```

File: docs/ops/deployment/native_kubernetes.md [hunk 3]
```diff
@@ -170,6 +170,34 @@ appender.console.layout.pattern = %d{yyyy-MM-dd HH:mm:ss,SSS} %-5p %-60c %x - %m
...
+
+Note: Only "local" is supported as schema for application mode. This assumes that the jar is located in the image, not the Flink client.
+
+Note: All the jars in the "$FLINK_HOME/usrlib" directory in the image will be added to user classpath.
+
+### Stop Flink Application
+
+When all the jobs in the application finished, the Flink cluster will be destroyed automatically. Also you could cancel all the Flink jobs to stop the application via cli or web dashboard.
```
Comment (sjwiesman) [inline docs/ops/deployment/native_kubernetes.md:196]:
```suggestion
When an application is stopped, all Flink cluster resources are automatically destroyed.
As always, Jobs may stop when manually canceled or, in the case of bounded Jobs, complete. 

```

**Comment by wangyang0918** [comment]
@sjwiesman Thanks a lot for your review, especially for the text. I have integrated your suggestions and make some small change. Adding a line `RUN mkdir -p $FLINK_HOME/usrlib` in Dockerfile since in the current Flink release image it does not exist.

I also create a ticket for the translation FLINK-17814. I will do it in the following days.

Please have another look.

**Comment by sjwiesman** [comment]
Fixed in 7723774a0402e10bc914b1fa6128e3c80678dafe

**Pull request closed without merging**